lib/prawn/attachment.rb

```diff
@@ -39,11 +39,9 @@ class Error < StandardError; end
     # hidden, then nil is returned.
     #
     def attach(src, options = {})
-      path = Pathname.new(src)
-
-      raise ArgumentError, "Data source can't be a directory" if path.directory?
+      raise ArgumentError, "Data source can't be a directory" if directory?(src)
```

Comment:
This is confusing. We should be testing for what we *want*, not what we *don't want*.

Reply:
I'm not entirely sure about this line, should we remove this check? Checking for `file?` is not the same

Reply:
You need to think of this more as "story telling". The basic objective with this entry point is to check if the use has provided a path to a file, or an IO object. Anything else is invalid. So start there:

```suggestion
      raise ArgumentError, "data source must be file path or io object" unless file?(src)
```

Reply:
Yeah, but `file?` won't be valid for this. I could leave `if directory?(src)` and change only the message

Reply:
Why not? That's my point. It's not **obvious** from the code what this method needs.